test/unit/rfc/paths_rule.cpp

```diff
@@ -190,13 +192,68 @@ class paths_rule_test
         bad<T>("/%25%2e");
     }
 
+    void
+    test_normalize()
+    {
+        {
+            auto check = [](
+                   string_view p,
+                   string_view e1,
+                   string_view e2) {
+                std::allocator<char> a{};
+                auto s1 = detail::remove_dot_segments(p, a, true);
+                BOOST_TEST(e1 == s1);
+
+                auto s2 = detail::remove_dot_segments(p, a, false);
+                BOOST_TEST(e2 == s2);
+
+                BOOST_TEST(detail::remove_dot_segments_compare(p, e1, true) == 0);
+                BOOST_TEST(detail::remove_dot_segments_compare(p, e2, false) == 0);
+            };
+
+            check("/a/b/c/./../../g", "/a/g", "/a/g");
+            check("/aa/bb/cc/./../../gg", "/aa/gg", "/aa/gg");
+            check("../a/b/c/./../../g", "a/g", "../a/g");
+            check("./a/b/c/./../../g", "a/g", "a/g");
+            check(".././a/b/c/./../../g", "a/g", "../a/g");
+            check("/a/b/../../g", "/g", "/g");
+            check("/a/b/../../../g", "/g", "/../g");
+            check("mid/content=5/../6", "mid/6", "mid/6");
+            check("mid/content=5/../6/.", "mid/6/", "mid/6/");
+            check("mid/content=5/../6/..", "mid/", "mid/");
+            check("/..", "/", "/..");
+            check(".", "", "");
+            check("..", "", "..");
+            check("", "", "");
+        }
+
+        // inequality
+        {
+            BOOST_TEST(detail::remove_dot_segments_compare(
+                "./a/b/c/./../../g", // "a/g"
+                "/a/b/../../../g",   // "/../g"
+                false) == +1);
```

Comment:
Isn't `+1` the same as `1` ?

Reply:
Yep. I just noticed people, including the pros, tend to use +1 in `compare` or `operator<=>` algorithms to make it more explicit:

https://github.com/boostorg/core/blob/dbefea96316465c501bcb853642cc855b03a462b/include/boost/core/detail/string_view.hpp#L589

Reply:
The problem with "+1" is that it is weird, and makes me waste time wondering if the code would work the same way if it was written as just "1". There is a time for "+1" which is in a lambda without a specified return type that wants to return an unsigned (or something like that). But this is not that so it should not get fancy.

I asked Peter why he used +1 in string_view and he said "just for looks." In other words there is no difference between +1 and 1 in this context.

Reply:
Yep. I knew all along it was just for looks. It looked better to me too. Anyway, I've changed it.